py_dataset/dataset.py

```diff
@@ -692,9 +645,9 @@ def repair(collection_name):
         return ''
     return error_message()
 
-def attach(collection_name, key, filenames = []):
+def attach(collection_name, key, semver, filenames = []):
```

Comment:
Could there be a default for semver of v0.0.0 (or whatever the command line version does?)